### PR TITLE
Remove Defense obstacles from coach node

### PR DESCRIPTION
**Pull request by petergarud**
## Description
Removed defense obstacles calculation from coach node and moved to Global State

## Associated / Resolved Issue
Resolves # https://app.clickup.com/t/86ayb2pbt

## Review Checklist

- [ ] **Docstrings**: All methods and classes should have the file appropriate docstrings which follow the guidelines in the ["Contributing" page](https://rj-rc-software.readthedocs.io/en/latest/contributing.html) of our docs.
- [ ] **Remove extra print statements**: Any print statements used for debugging should be removed
- [ ] **Tag reviewers**: Tag some people for review and ping them on Slack

## (Optional) Sub-issues (for drafts)
_Note: if you find yourself breaking this PR into many smaller features, it may make sense to break up the PR into logical units based on these features._
- [ ] Step 1
- [ ] Step 2

### CODE REVIEW

File: soccer/src/soccer/planning/planner_node.hpp [hunk 1]
```diff
@@ -107,14 +110,55 @@ class GlobalState {
     rclcpp::Subscription<rj_geometry_msgs::msg::ShapeSet>::SharedPtr def_area_obstacles_sub_;
     rclcpp::Subscription<rj_msgs::msg::WorldState>::SharedPtr world_state_sub_;
     rclcpp::Subscription<rj_msgs::msg::CoachState>::SharedPtr coach_state_sub_;
+    rclcpp::Subscription<rj_msgs::msg::FieldDimensions>::SharedPtr field_dimensions_sub_;
 
     PlayState last_play_state_ = PlayState::halt();
+    FieldDimensions current_field_dimensions_;
+    bool have_field_dimensions_ = false;
```
Comment (rishiso) [inline soccer/src/soccer/planning/planner_node.hpp]: this could just be `bool have_field_dimensions_;`

File: rj_gameplay/stp/pylint_stp.py
Comment (rishiso) [inline rj_gameplay/stp/pylint_stp.py:1]: Uncommit this file please

File: rj_common/include/rj_common/network.hpp
Comment (rishiso) [inline rj_common/include/rj_common/network.hpp:1]: Uncommit this file please

File: .gitignore
Comment (rishiso) [inline .gitignore:1]: This change is fine, but please put this in a separate PR

**Review by p-nayak11 — APPROVED** [review]

**Comment by p-nayak11** [comment]
Merged into #2107
